include/dr/mhp/containers/segments.hpp

```diff
@@ -0,0 +1,174 @@
+
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "index.hpp"
+
+namespace dr::mhp {
+
+using key_type = index<>;
+
+template <typename DV> class d_segment_iterator;
```

Comment:
looks the same as dv_segment_iterator, dv_segment_reference
what is different? I don't see any specific 2d code here
Can it be shared? if so please just use dv_segment_iterator and dv_segment_reference

Reply:
TODO